### ISSUE TITLE
Locate is slow, difficult or fails entirely

### ISSUE THREAD

**Issue by privatemaker** [organicmaps/organicmaps#5669]
**Describe the issue**

Clicking the `Locate` icon button, OMaps fails to determine location of user.

This problem is acute and makes OMaps routing highly unreliable and/or entirely unsuable.

Related #3450

**Steps to reproduce**

In Android Location permission settings, OMaps has the same state granted as other map apps:

```
Allow only while using the app   [Y]
```

0. Use device with Calyx, Lingeage, or Graphene OS without Google Services
1. Place phone in airplane mode. Disable Location access.
2. Click the `Locate` icon button 
3. The `Locating` radar icon starts spinning.

After multiple minutes in the `Locating` state, a pop-up asks `Continue detecting your location?`
This icon spinning state and failure persists through the following:

- Turning off Airplane mode
- Enabling device Location
- Quitting app 
- Force stopping app
- Toggling `Disable / Enable` app

After all attempts at changing the phones state, Organic Maps fails to resolve any result or activate. 

In none of the above states, does OMaps trigger the `Accessing Location` green icon in the phone's topbar, which other mapping apps with the same permissions and user conditions do. 

<img src="https://github.com/organicmaps/organicmaps/assets/49612519/3e7a95c0-d025-4520-bf68-c13da2d4e09e" width="475">

OsmAnd~ and Transportr both work fine and trigger the `GPS Locating` UI widget of the OS, and properly show me on a map after updating the GPS data. Once this has occurred Organic Maps can also locate me successfully. 

**Expected behavior**

OMaps should properly activate GPS, download GPS data, and locate user.

**Suggested Improvements**

- If unable to access location, at bare minimum inform user why it cannot (Location settings, etc.), Transportr displays this message:

<img src="https://github.com/organicmaps/organicmaps/assets/49612519/980e12a3-e23f-42bd-a680-eb6fd69c32e4" width="475">

- Popular FOSS apps Transportr and OsmAnd~ both display following status messages to user:

<img src="https://github.com/organicmaps/organicmaps/assets/49612519/76848299-ccaa-4b76-912c-ee4402a6afe6" width="475">

<img src="https://github.com/organicmaps/organicmaps/assets/49612519/205cbf1f-c1c7-4b62-9826-90bdcf92fcd3" width="475">

- Allow users to manually specify their location. 

SCORE: 4

**System information:**

- Operating system and its version: Android 13 / Graphene OS / Calyx OS
- Organic Maps version: 2023.07.13-FDroid

**Comment by biodranik** [comment]
@rtsisyk this is something that was already reported by some other users.

**Comment by privatemaker** [comment]
> ...at bare minimum inform user why it cannot (Location settings, etc.),

I made a mistake above in my testing- I did not trigger `Please Enable location Services` dialogue because I only tested with `Location: enabled` i've updated issue for ref.

However, I do not consider this a duplicate of the potential `fused` issue which is technical  AFAICT, while above is also a UX issue.

1. The notification `Position not yet known` other maps apps offer shows immediately, whereas in OMaps it acts as though it is trying for minutes, and then asks to keep trying. This a UX failure to accurately communicate a failure mode which feels absolute.

2. Allowing users to manually update their location is useful IMHO.

**Comment by biodranik** [comment]
@privatemaker am I correctly understood that now OM does not detect that location is disabled in the device's settings? If yes, then it is a bug, not a UX issue. The system should report it immediately to the app.

**Comment by privatemaker** [comment]
@biodranik no, when location is disabled in devices settings, it pops this up:

<img src="https://github.com/organicmaps/organicmaps/assets/49612519/603db08a-b583-43c9-ba51-685f84e3361a" width="350">

My issue is about clicking the locate button *when* Location is enabled, and OMaps not triggering the green location activated icon in the top right. Instead it takes forever and then shows this:

<img src="https://github.com/organicmaps/organicmaps/assets/49612519/1fcca950-5bf4-4c61-b31d-c142c814a6fe" width="350">

This very well might be a duplicate issue of #4223

**Comment by privatemaker** [comment]
If the #4223 fix by [@rtsisyk](https://github.com/rtsisyk) does indeed solve this, I will verify once new version is available in F-Droid after testing in Lineage, Graphene, and Calyx.

**Comment by privatemaker** [comment]
hooray 🎉 it is resolved the green `Accessing Location` icon appears now! Nice work @rtsisyk 🥳
